R/utils_groupcomparison_plots.R

```diff
@@ -55,21 +57,55 @@
 }
 
 
+colMax <- function(data) sapply(data, max, na.rm = TRUE)
+colMin <- function(data) sapply(data, min, na.rm = TRUE)
+
 #' Create heatmap
 #' @param input data.table
 #' @inheritParams groupComparisonPlots
 #' @keywords internal
-.makeHeatmap = function(input, my.colors, my.breaks, x.axis.size, y.axis.size) {
+.makeHeatmap = function(input, my.colors, my.breaks, x.axis.size, y.axis.size, height) {
     par(oma = c(3, 0, 0, 4))
-    heatmap.2(as.matrix(input),
-              col = my.colors,
-              Rowv = FALSE, Colv = FALSE,
-              dendrogram = "none", breaks = my.breaks,
-              trace = "none", na.color = "grey",
-              cexCol = (x.axis.size / 10), 
-              cexRow = (y.axis.size / 10),
-              key = FALSE,
-              lhei = c(0.1, 0.9), lwid = c(0.1, 0.9)) 
+
+    label_formatter <- list(
+        title = "",
+        # titlefont = f1,
+        showticklabels = TRUE,
+        tickangle = 45,
+        # tickfont = f2,
+        exponentformat = "E")
+    
+    # adjust my.breaks
+    x = my.breaks
+    dltx <- diff(x)[1]
+    x <- sort(c(x,-dltx/16,dltx/16))
+    x <- x[x!=0]
+    x.resc <- (x-min(x))/(max(x)-min(x))
+    
+    # get color scale
+    cols = my.colors
+    colorScale <- data.frame(
+        z = c(0,rep(x.resc[2:(length(x.resc)-1)],each=2),1),
+        col=rep(cols,each=2)
+    )
+    
+    heatmap_plot = plot_ly(z = as.matrix(input),
+            zmin=x[1],
+            zmax=x[length(x)],
+            x = colnames(input),
+            xgap = 0,
+            y = rownames(input),
+            ygap = 0,
+            type = "heatmap",
+            showlegend=FALSE, showscale=FALSE,
+            colorscale = colorScale,
+            colorbar=list(ypad = 520, tick0=x[1], dtick=dltx,len=1,orientation="h")
+            ) %>%
+        plotly::layout(xaxis = label_formatter, 
+               plot_bgcolor  = "grey"
+        )
+    print(heatmap_plot)
+    heatmap_plot
```

Comment:
do we need both?

Reply:
Commit [remove debug](https://github.com/Vitek-Lab/MSstats/pull/107/commits/15d34012bcc14e60eed2be134e30f16a53d13699)